src/blockdata/transaction.rs

```diff
@@ -608,6 +608,22 @@ impl Decodable for Transaction {
     }
 }
 
+/// Error occuring when parsing a SigHashType
+#[derive(Debug, PartialEq)]
```

Comment:
Can we add more derives to this? As decided in #555 that it makes sense to always derive if it makes sense. I think we need Ord, Eq, PartialOrd, Hash, Clone, Copy.

Reply:
Sure thing but out of curiosity: why `Ord` and `PartialOrd` for an error?

Reply:
Even I was opposed to it initially. But other reviewers wanted support for BtreeSet, BtreeMap for objects that might use Error types.  
https://github.com/rust-bitcoin/rust-bitcoin/issues/555#issuecomment-764958577
https://github.com/rust-bitcoin/rust-bitcoin/pull/511#issuecomment-764969363